### DISCUSSION TITLE
Resizing Folium Map Frame?

### DISCUSSION THREAD

**Discussion by peterkobylarz** [earthlab-education/Earth-Analytics-2023-01-Intro#319] [Q&A03 Python]
I was trying to reduce the size of the Folium map from the week 3 coding exercise. I was able to adjust the height and width of the map itself using the height and width properties:

```
m = folium.Map(
    # Location to display
    location=(sg_lat, sg_lon),
    # Turns off annoying zooming while trying to scroll to the next cell
    scrollWheelZoom=False,
    tiles = "https://server.arcgisonline.com/ArcGIS/rest/services/NatGeo_World_Map/MapServer/tile/{z}/{y}/{x}",
    attr = 'Tiles &copy; Esri &mdash; National Geographic, Esri, DeLorme, NAVTEQ, UNEP-WCMC, USGS, NASA, ESA, METI, NRCAN, GEBCO, NOAA, iPC',
    zoom_start = 9,
    height = 1000,
    width = 1000
)
```
However, it seems like this only resized the map itself. There appears to be some sort of frame containing it which remained the same size, resulting in a lot of white space.

Does anyone have any ideas how to adjust the overall frame of the Folium map? Is there a property to specify in the Folium parameters or would this have to be controlled by HTML or CSS?

**Comment by jsmi374** [comment]
Hey Peter, I looked into this after class, and found some lines of code that lets you change the size of the map, and eliminates the trailing whitespace. Here is a screen shot of the implemented code.  @peterkobylarz  @nquarder 

![Change_map_Size](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/76793232/ccdb2329-a626-426c-a829-400fe47b490a)

**Reply by jsmi374** [reply]
f=folium.Figure(width = 1000, height = 500)
m = folium.Map(
    # Location to display
    location=(sg_lat, sg_lon),
    # Turns off annoying zooming while trying to scroll to the next cell
    scrollWheelZoom=False,
    tiles = "https://server.arcgisonline.com/ArcGIS/rest/services/NatGeo_World_Map/MapServer/tile/{z}/{y}/{x}",
    attr = 'Tiles &copy; Esri &mdash; National Geographic, Esri, DeLorme, NAVTEQ, UNEP-WCMC, USGS, NASA, ESA, METI, NRCAN, GEBCO, NOAA, iPC',
    zoom_start = 9,
    height = 1000,
    width = 1000
).add_to(f)

**Reply by jsmi374** [reply]
just edits to the first and last line of your code  provided above.

**Reply by peterkobylarz** [reply]
Thanks James. I just tested this out and I'm still running into the issue where the map is just resizing within the larger Folium frame which leaves a bunch of whitespace to the right and below. I just decided to leave it as is for my portfolio assignment.

**Comment by eculler** [comment]
@peterkobylarz I'm not aware of any way to do this using the folium library (although your and @jsmi374 's solution works for resizing the map itself as you say). It's possible that you can reduce the frame size using the `IPython.core.display` `HTML` class  -- See https://stackoverflow.com/questions/25698448/how-to-embed-html-into-ipython-output

Another approach to try would be saving your folium map to a file and then displaying it in Markdown/HTML directly. That could be a nice way to have more control over an output notebook.
